### PR TITLE
Checkstyle: Remove empty Javadoc tags

### PR DESCRIPTION
**Pull request by ssoloff**
This PR fixes violations of the Checkstyle rule NonEmptyAtclauseDescription.

**This change only affects Javadocs; no code was modified.**

Within Eclipse, these empty tags offer no additional information.  Eclipse's Content Assist feature automatically populates parameters and thrown exceptions in the method signature regardless of whether or not these tags are present.

However, when generating Javadocs using the `javadoc` tool, this change will cause some visible differences in the resulting output.  The removal of empty `@param` tags will cause those entries to be removed from the **Parameters** section, but they will still be present in the method signature.  However, the removal of empty `@throws` tags causes no change, as the exception (with empty description) continues to be present  in the **Throws** section.

For example, for the following change to `UnitSupportAttachment`:

```diff
diff --git a/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java b/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java
index 8b85089..92d1a80 100644
--- a/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java
+++ b/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java
@@ -246,9 +246,6 @@ public class UnitSupportAttachment extends DefaultAttachment {
 
   /**
    * Adds to, not sets. Anything that adds to instead of setting needs a clear function as well.
-   *
-   * @param names
-   * @throws GameParseException
    */
   @GameProperty(xmlProperty = true, gameProperty = true, adds = true)
   public void setPlayers(final String names) throws GameParseException {
```

The generated Javadocs for the `setPlayers(String)` method will change from

javadocs-before-refactoring

to

javadocs-after-refactoring

I don't believe this should be an issue, as those removed rows did not offer any useful information, but please advise if someone disagrees.  Note that Eclipse's Javadoc view/hover is unaffected--it will generate the same output (i.e. a parameter list with empty descriptions) regardless of whether or not the `@param` tags are present.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-checkstyleMainMaxWarnings=7798
-checkstyleTestMaxWarnings=381
+checkstyleMainMaxWarnings=7420
```
Comment (DanVanAtta) [inline gradle.properties]: Is it correct that with 529 lines removed, we removed 7000 some warnings?

**Comment by DanVanAtta** [comment]
Updates look good, double check on the new warning count please.

File: gradle.properties (hunk 1 above)
Comment (DanVanAtta) [inline gradle.properties]: Ah, nevermind, these are two different values, I misread it earlier.

**Comment by DanVanAtta** [comment]
Nevermind on double check, I misread. 
Just the merge conflict as far as I can see.

**Comment by ssoloff** [comment]
Rebased against `master` at 356efbf (1.9.0.0.4056).

**Pull request merged**